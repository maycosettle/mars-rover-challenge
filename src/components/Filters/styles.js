import classNames from 'classnames';

const useStyles = () => {
  const container = 'flex items-center justify-center mt-6 p-2';

  return {
    container: 'grid grid-cols-7',
    checkbox: 'mt-8 p-2',
    padding: 'p-2',
    search: container,
    icon: '-ml-0.5 mr-2 h-4 w-4',
    bookmark: classNames(container, 'border-l'),
  };
};

export default useStyles;
