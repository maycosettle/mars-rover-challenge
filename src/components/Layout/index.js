export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
