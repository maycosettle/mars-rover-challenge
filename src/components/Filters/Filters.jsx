import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import { BookmarkIcon, SearchIcon } from '@heroicons/react/solid';

import { ROVER_OPTIONS } from 'utils/constants';

import { Button } from 'components/Button';
import { Checkbox } from 'components/Checkbox';
import { Select } from 'components/Select';

import useStyles from './styles';

const Filters = ({
  solCheckbox,
  earthCheckbox,
  selectedRover,
  cameraOptions,
  selectedCamera,
  bookmarkOptions,
  setSolCheckbox,
  selectedBookmark,
  setEarthCheckbox,
  handleCamera,
  handleRover,
  handleSearch,
  handleOpen,
  handleBookmark,
}) => {
  const disableSearch = (solCheckbox || earthCheckbox) && !isEmpty(selectedRover) && !isEmpty(selectedCamera);

  const styles = useStyles();

  return (
    <div className={styles.container}>
      <div className={styles.checkbox}>
        <Checkbox
          label="Sol date"
          description="2890"
          checked={solCheckbox}
          onChange={() => setSolCheckbox(!solCheckbox)}
        />
      </div>
      <div className={styles.checkbox}>
        <Checkbox
          label="Earth day"
          description="2020-09-22"
          checked={earthCheckbox}
          onChange={() => setEarthCheckbox(!earthCheckbox)}
        />
      </div>
      <div className={styles.padding}>
        <Select
          label="Rover"
          options={ROVER_OPTIONS}
          selected={selectedRover}
          setSelected={handleRover}
          placeholder="Select a rover"
        />
      </div>
      <div className={styles.padding}>
        <Select
          label="Rover Camera"
          options={cameraOptions}
          selected={selectedCamera}
          setSelected={handleCamera}
          placeholder="Select a camera"
        />
      </div>
      <div className={styles.search}>
        <Button
          name="Search"
          icon={<SearchIcon className={styles.icon} aria-hidden="true" />}
          onClick={handleSearch}
          disabled={!disableSearch}
        />
      </div>
      <div className={styles.bookmark}>
        <Button
          name="Add bookmark"
          icon={<BookmarkIcon className={styles.icon} aria-hidden="true" />}
          onClick={handleOpen}
        />
      </div>
      <div className={styles.padding}>
        <Select
          label="Bookmarks"
          options={bookmarkOptions}
          selected={selectedBookmark}
          setSelected={handleBookmark}
          placeholder="Bookmarks"
        />
      </div>
    </div>
  );
};

Filters.propTypes = {
  solCheckbox: PropTypes.bool,
  earthCheckbox: PropTypes.bool,
  selectedRover: PropTypes.object,
  cameraOptions: PropTypes.array,
  selectedCamera: PropTypes.object,
  bookmarkOptions: PropTypes.array,
  selectedBookmark: PropTypes.object,
  setSolCheckbox: PropTypes.func,
  setEarthCheckbox: PropTypes.func,
  handleCamera: PropTypes.func,
  handleRover: PropTypes.func,
  handleSearch: PropTypes.func,
  handleOpen: PropTypes.func,
  handleBookmark: PropTypes.func,
};

Filters.defaultProps = {
  solCheckbox: false,
  earthCheckbox: false,
  selectedRover: {},
  cameraOptions: [],
  selectedCamera: {},
  bookmarkOptions: [],
  selectedBookmark: '',
  setSolCheckbox: () => {},
  setEarthCheckbox: () => {},
  handleCamera: () => {},
  handleRover: () => {},
  handleSearch: () => {},
  handleOpen: () => {},
  handleBookmark: () => {},
};

export default Filters;
