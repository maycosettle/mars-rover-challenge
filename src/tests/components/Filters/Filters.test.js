import { shallow } from 'enzyme';
import { Filters } from 'components/Filters';

describe('tests on <Filters />', () => {
  const wrapper = shallow(<Filters />);

  test('Should render Filters correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
