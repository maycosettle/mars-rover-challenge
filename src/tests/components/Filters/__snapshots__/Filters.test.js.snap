// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Filters /> Should render Filters correctly 1`] = `
<div
  className="grid grid-cols-7"
>
  <div
    className="mt-8 p-2"
  >
    <Checkbox
      checked={false}
      description="2890"
      label="Sol date"
      onChange={[Function]}
    />
  </div>
  <div
    className="mt-8 p-2"
  >
    <Checkbox
      checked={false}
      description="2020-09-22"
      label="Earth day"
      onChange={[Function]}
    />
  </div>
  <div
    className="p-2"
  >
    <Select
      label="Rover"
      options={
        Array [
          Object {
            "id": "Curiosity",
            "name": "Curiosity",
          },
          Object {
            "id": "Opportunity",
            "name": "Opportunity",
          },
          Object {
            "id": "Spirit",
            "name": "Spirit",
          },
        ]
      }
      placeholder="Select a rover"
      selected={Object {}}
      setSelected={[Function]}
    />
  </div>
  <div
    className="p-2"
  >
    <Select
      label="Rover Camera"
      options={Array []}
      placeholder="Select a camera"
      selected={Object {}}
      setSelected={[Function]}
    />
  </div>
  <div
    className="flex items-center justify-center mt-6 p-2"
  >
    <Button
      bgColor="primary"
      className=""
      color="white"
      disabled={true}
      icon={
        <SearchIcon
          aria-hidden="true"
          className="-ml-0.5 mr-2 h-4 w-4"
        />
      }
      name="Search"
      onClick={[Function]}
      rounded="full"
      transparent={false}
      type="button"
    />
  </div>
  <div
    className="flex items-center justify-center mt-6 p-2 border-l"
  >
    <Button
      bgColor="primary"
      className=""
      color="white"
      disabled={false}
      icon={
        <BookmarkIcon
          aria-hidden="true"
          className="-ml-0.5 mr-2 h-4 w-4"
        />
      }
      name="Add bookmark"
      onClick={[Function]}
      rounded="full"
      transparent={false}
      type="button"
    />
  </div>
  <div
    className="p-2"
  >
    <Select
      label="Bookmarks"
      options={Array []}
      placeholder="Bookmarks"
      selected=""
      setSelected={[Function]}
    />
  </div>
</div>
`;
